metrics/stackdriver_exporter.go

```diff
@@ -63,10 +65,18 @@ func newOpencensusSDExporter(o stackdriver.Options) (view.Exporter, error) {
 // TODO should be properly refactored to be able to inject the getMonitoredResourceFunc function.
 // 	See https://github.com/knative/pkg/issues/608
 func newStackdriverExporter(config *metricsConfig, logger *zap.SugaredLogger) (view.Exporter, error) {
-	gm := gcpMetadataFunc()
+	gm := getGCPMetadata(config)
```

Comment:
Reading this code, I think it's easy to conclude that getGCPMetadata is returning the metadata from GCP, and not the result of merging with config. Consider a more descriptive name like `getMergedGCPMetadata()`, or refactoring to something like `mergeGCPMetadata(retrieveGCPMetadata(), config)`.